Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.cpp

```diff
@@ -294,27 +256,34 @@ Ref<CoordinatedTileBuffer> SkiaPaintingEngine::paintLayer(const GraphicsLayer& l
 
 Ref<CoordinatedTileBuffer> SkiaPaintingEngine::postPaintingTask(const GraphicsLayer& layer, RenderingMode renderingMode, const IntRect& dirtyRect, bool contentsOpaque, float contentsScale)
 {
+    auto buffer = createBuffer(renderingMode, dirtyRect.size(), contentsOpaque);
+    buffer->beginPainting();
+
     WTFBeginSignpost(this, RecordTile);
-    auto displayList = recordDisplayList(renderingMode, layer, dirtyRect, contentsOpaque, contentsScale);
+    auto* recordingCanvas = m_pictureRecorder.beginRecording(dirtyRect.width(), dirtyRect.height());
+    GraphicsContextSkia recordingContext(*recordingCanvas, renderingMode, RenderingPurpose::LayerBacking, nullptr, GraphicsContextSkia::ContextType::RecordingContext);
+    paintIntoGraphicsContext(layer, recordingContext, dirtyRect, contentsOpaque, contentsScale);
+    auto picture = m_pictureRecorder.finishRecordingAsPicture();
```

Comment:
I have another idea. What if instead of adding a context type construction parameter and stealing the fences layer, we don't add beginRecording/endRecording? In the beginRecording we set an internal flag in the context to indicate we are recording, and endRecording can return the fences.

Reply:
That is perfect, I did that.